src/popup/utils/helper.js

```diff
@@ -87,6 +87,7 @@ export const detectConnectionType = (port) => {
 
 export const fetchJson = async (...args) => {
   const response = await fetch(...args);
+  if (response.status === 204) return new Promise(() => {});
```

Comment:
Suggestions:
1. Move this to helper TS file to have proper return type
2. Maybe in this case this function should throw error? Or at least return null, as `null` is a thing that means "nothing".